test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,354 @@
+    validate_state(
+        mz,
+        reached_index,
+        must_exceed_reached_index=False,
+        timeout_in_sec=30,
+    )
+
+    # only request this index because the previous validation succeeded / did not block
+    stalled_index = get_current_counter_index(mz)
+
+    print("Awaiting live data due to node recovery...")
+    # expect live data to be delivered at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        stalled_index,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+    recovered_index = get_current_counter_index(mz)
+
+    mz.revive_suspended_k8s_node(suspended_node_name)
+
+    # expect no issues after node recovery
+    validate_state(
+        mz,
+        recovered_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=10,
+    )
+
+
+# @pytest.mark.long
+# @unittest.skip("Requires #20299 to work")
+def test_unreplicated_compute_cluster_on_failing_node(
+    mz: MaterializeApplication,
+) -> None:
+    """
+    An unreplicated compute cluster is on the failed node. Queries of indexes on the compute cluster should fail, but
+    resolve within a minute or two.
+    """
+    compute_cluster = default_compute_cluster()
+    storage_cluster = default_storage_cluster()
+
+    populate(mz, compute_cluster, storage_cluster)
+    reached_index = get_current_counter_index(mz)
+
+    suspended_node_name = suspend_node_of_replica(
+        mz, compute_cluster, compute_cluster.replica_definitions[0]
+    )
+
+    print("Awaiting node recovery...")
+    # expect (live) data to be delivered after at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        reached_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+    recovered_index = get_current_counter_index(mz)
+
+    mz.revive_suspended_k8s_node(suspended_node_name)
+
+    # expect no issues after node recovery
+    validate_state(
+        mz,
+        recovered_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=10,
+    )
+
+
+# @pytest.mark.long
+def test_replicated_compute_cluster_on_failing_node(mz: MaterializeApplication) -> None:
+    """
+    A replicated compute cluster is on the failed node. Queries of indexes on the compute cluster should experience no
+    disruption in latency, thanks to the second replica.
+    """
+    compute_cluster = default_compute_cluster()
+    compute_cluster.replica_definitions.append(
+        ReplicaDefinition(compute_cluster.name, 2, "2")
+    )
+    storage_cluster = default_storage_cluster()
+
+    populate(mz, compute_cluster, storage_cluster)
+    reached_index = get_current_counter_index(mz)
+
+    suspended_node_name = suspend_node_of_replica(
+        mz, compute_cluster, compute_cluster.replica_definitions[1]
+    )
+
+    print("Awaiting node recovery...")
```

Comment:
You are not awaiting for anything here, I think this `print` needs to be removed.